src/float-counts-estimate.cc

```diff
@@ -321,7 +321,7 @@ class FloatCountsEstimator {
     const FloatLmStateWork &work = work_[history_length];
     assert(work.counts.size() == lm_state.counts.size());
 
-    float old_total = lm_state.total,
+    double old_total = lm_state.total,
```

Comment:
Also, you have various other bug fixes-- if you have time it would be great if you could put all those bug fixes in a separate PR from this one, so I can merge it before this PR is merged.